python/tvm/contrib/hexagon/session.py

```diff
@@ -86,7 +88,15 @@ def __enter__(self):
                     self._rpc_receive_buffer_size_bytes,
                 ],
             )
-            self.device = self._rpc.hexagon(0)
+            if self._session_name == "cpu-rpc":
```

Comment:
Proposed option:  Change `device` to be lazily generated in a `@property`, rather than set during the `__enter__` method.

Reply:
This will require some of the `if self._device is None` checks to be replaced with `if self._rpc is None`.

Reply:
done.